src/Examine.Lucene/Search/LuceneSearchQueryBase.cs

```diff
@@ -505,6 +506,14 @@ public override string ToString()
             return $"{{ Category: {Category}, LuceneQuery: {Query} }}";
         }
 
+        public abstract IFacetQueryField Facet(string field);
```

Comment:
Wondering on the naming here. These methods are essentially adding faceting capabilities to the result (?) so maybe this could be more of a builder syntax like `WithFacet` - would that makes sense?

Reply:
Great naming suggestion!